ecl2df/compdat.py

```diff
@@ -241,6 +255,76 @@ def unrolldf(dframe, start_column="K1", end_column="K2"):
     return unrolled
 
 
+def applywelopen(compdat_df, welopen_df):
+    """Apply WELLOPEN actions to the COMPDAT dataframe.
+
+    Example: COMPDAT and WELOPEN keyword::
+
+      COMPDAT
+        'OP1' 33 44 10 11 'OPEN' /
+      /
+      WELOPEN
+        'OP1' SHUT /
+      /
+
+    will be equal to defining the connections SHUT within COMPDAT.
+
+    Args:
+        compdat_df (pd.DataFrame): Dataframe with unrolled COMPDAT data
+        welopen_df (pd.DataFrame): Dataframe with WELOPEN actions
+
+    Returns:
+        pd.Dataframe, compdat_df now including WELOPEN actions
+
+    """
+    welopen_df = welopen_df.astype(object).where(pd.notnull(welopen_df), None)
+    for _, row in welopen_df.iterrows():
+        if row["I"] and row["J"] and row["K"]:
+            previous_state = compdat_df[
+                (compdat_df["WELL"] == row["WELL"])
+                & (compdat_df["DATE"] <= row["DATE"])
+                & (compdat_df["I"] == row["I"])
+                & (compdat_df["J"] == row["J"])
+                & (compdat_df["K1"] == row["K"])
+                & (compdat_df["K2"] == row["K"])
+            ].drop_duplicates(subset=["I", "J", "K1", "K2"], keep="last")
+        elif row["C1"] or row["C2"]:
+            raise ValueError(
+                "Lumped connections are not supported in the a WELOPEN keyword."
+            )
+        elif not (row["I"] and row["J"] and row["K"]):
+            previous_state = compdat_df[
+                (compdat_df["WELL"] == row["WELL"])
+                & (compdat_df["DATE"] <= row["DATE"])
+            ].drop_duplicates(subset=["I", "J", "K1", "K2"], keep="last")
+        else:
+            raise ValueError(
```

Comment:
This situation more warrants a hard error, so catching all ValueError's  will probably be to  broad?

Reply:
This is a value error as the combination of values don't make sense (I and J is given, but not K). Of course one could write a error catching for all individual values... But I guess this will do.